Lib/test/test_threading.py

```diff
@@ -961,6 +962,37 @@ def import_threading():
         self.assertEqual(out, b'')
         self.assertEqual(err, b'')
 
+    def test_thread_state_name(self):
+        def target():
+            thread = threading.current_thread()
+
+            self.assertIs(thread.name, _testcapi.get_thread_name(thread._tstate))
+
+            thread.name = "new-thread-name"
+            self.assertIs(thread.name, _testcapi.get_thread_name(thread._tstate))
+
+        thread = threading.Thread(target=target, name="test-thread-name")
+        thread.start()
+
+        main_thread = threading.current_thread()
+        self.assertIs(main_thread.name, _testcapi.get_thread_name(main_thread._tstate))
+
+        thread.join()
+
+        # Check we can still set the name after the thread terminated
+        thread.name = "foo"
```

Comment:
This could use some more tests checking that the name you set is actually reflected.

Reply:
This was meant to catch any potential segfaults, since once the thread terminates there isn't a thread state anymore.